Mathlib/Computability/ContextFreeGrammar.lean

```diff
@@ -31,20 +31,20 @@ structure ContextFreeGrammar.{uN,uT} (T : Type uT) where
   rules : List (ContextFreeRule T NT)
 
 universe uT uN
-variable {T : Type uT} {N : Type uN}
+variable {T : Type uT}
 
 /-- One application of single context-free rule. -/
-inductive ContextFreeRule.RewritesTo (r : ContextFreeRule T N) :
+inductive ContextFreeRule.Rewrites {N : Type uN} (r : ContextFreeRule T N) :
```

Comment:
What's the advantage of moving `N` into the declaration here? You have four delcarations in a row that use it, so it's a bit easier to read on the `variable` line

Reply:
The advantage is that now `N` is no longer in the context in the whole rest of the file when `N` must not be used anymore (because `g.NT` dictates the nonterminal type).

Reply:
I think a `section` around the `ContextFreeRule` lemmas would probably be a good solution too

Reply:
OK, I'll make a section.